azure/table_azure_firewall.go

```diff
@@ -261,21 +261,24 @@ func ipConfigurationData(ctx context.Context, d *transform.TransformData) (inter
 	data := d.HydrateItem.(network.AzureFirewall)
 
 	var output []map[string]interface{}
-	for _, firewall := range *data.AzureFirewallPropertiesFormat.IPConfigurations {
-		objectMap := make(map[string]interface{})
-		if firewall.AzureFirewallIPConfigurationPropertiesFormat.PrivateIPAddress != nil {
-			objectMap["privateIPAddress"] = firewall.AzureFirewallIPConfigurationPropertiesFormat.PrivateIPAddress
-		}
-		if firewall.AzureFirewallIPConfigurationPropertiesFormat.PublicIPAddress != nil {
-			objectMap["publicIPAddress"] = firewall.AzureFirewallIPConfigurationPropertiesFormat.PublicIPAddress
-		}
-		if firewall.AzureFirewallIPConfigurationPropertiesFormat.Subnet != nil {
-			objectMap["subnet"] = firewall.AzureFirewallIPConfigurationPropertiesFormat.Subnet
-		}
-		if firewall.AzureFirewallIPConfigurationPropertiesFormat.ProvisioningState != "" {
-			objectMap["provisioningState"] = firewall.AzureFirewallIPConfigurationPropertiesFormat.ProvisioningState
+	if data.AzureFirewallPropertiesFormat.IPConfigurations != nil {
```

Comment:
```suggestion
// Add a check for AzureFirewallPropertiesFormat.IPConfigurations data to ensure that
// it is not null to avoid panic errors
	if data.AzureFirewallPropertiesFormat.IPConfigurations != nil {
```